src/Models/HydrostaticFreeSurfaceModels/split_explicit_free_surface.jl

```diff
@@ -272,38 +292,21 @@ function SplitExplicitSettings(FT::DataType=Float64;
 
         Δx⁻² = topology(grid)[1] == Flat ? 0 : 1 / minimum_xspacing(grid)^2
         Δy⁻² = topology(grid)[2] == Flat ? 0 : 1 / minimum_yspacing(grid)^2
-        Δs = sqrt(1 / (Δx⁻² + Δy⁻²))
+        Δs   = sqrt(1 / (Δx⁻² + Δy⁻²))
 
         wave_speed = sqrt(gravitational_acceleration * grid.Lz)
         
-        Δtᴮ = cfl * Δs / wave_speed
+        Δtᴮ = FT(cfl * Δs / wave_speed)
+        if substeps isa AdaptiveSubsteps
+            return SplitExplicitSettings(AdaptiveSubsteps(Δtᴮ, barotropic_averaging_kernel), 
+                                         timestepper)
+        end
         substeps = ceil(Int, 2 * max_Δt / Δtᴮ)
     end
 
-    τᶠ = range(0, 2, length = substeps+1)
-    Δτ = τᶠ[2] - τᶠ[1]
-
-    averaging_weights = FT.(barotropic_averaging_kernel.(τᶠ[2:end]))
-    idx = searchsortedlast(averaging_weights, 0, rev=true)
-    substeps = idx
-
-    averaging_weights = averaging_weights[1:idx]
-    mass_flux_weights = similar(averaging_weights)
-
-    M = searchsortedfirst(τᶠ, 1) - 1
-
-    averaging_weights ./= sum(averaging_weights)
-
-    for i in substeps:-1:1
-        mass_flux_weights[i] = 1 / M * sum(averaging_weights[i:substeps]) 
-    end
-
-    mass_flux_weights ./= sum(mass_flux_weights)
+    Δτ, averaging_weights = weights_from_substeps(FT, substeps, barotropic_averaging_kernel)
 
-    return SplitExplicitSettings(substeps,
-                                 averaging_weights,
-                                 mass_flux_weights,
-                                 Δτ,
+    return SplitExplicitSettings(FixedSubsteps(Δτ, averaging_weights),
                                  timestepper)
```

Comment:
The above can be

```julia
if substeps isa AdaptiveSubsteps
    substeps = AdaptiveSubsteps(Δtᴮ, barotropic_averaging_kernel)
else
    substeps = FixedSubsteps(Δτ, averaging_weights)
end

return SplitExplicitSettings(substeps, timestepper)
```

PS why do we have a distinct type for "Settings" ? This code is quite different from most of the other code; I think following the tyle of the rest of the code, `substeps` and `timestepper` would simply be properties of `SplitExplicitFreeSurface`. Not sure the advantage of the `Settings` type.

Reply:
It's to allow initialization with different settings (aka `cfl` and `grid` rather than `substeps`).